5.sc_Morphology_Logistic_Regression/0.split_data/scripts/split_data.py

```diff
@@ -0,0 +1,240 @@
+    cell_type = config["logistic_regression_params"]["cell_type"]
+    print(aggregation, nomic, cell_type)
+
+
+# In[5]:
+
+
+path = pathlib.Path(f"../../data/{cell_type}_preprocessed_sc_norm.parquet")
+
+data_df = pq.read_table(path).to_pandas()
+
+
+# In[6]:
+
+
+if nomic == True:
+    nomic_df_path = pathlib.Path(
+        f"../../2.Nomic_nELISA_Analysis/Data/clean/Plate2/nELISA_plate_430420_{cell_type}.csv"
+    )
+    df_nomic = pd.read_csv(nomic_df_path)
+    # drop columns that contain [pgML]
+    df_nomic = df_nomic.drop(
+        columns=[col for col in df_nomic.columns if "[pgML]" in col]
+    )
+    # drop first 25 columns
+    df_nomic = df_nomic.drop(columns=df_nomic.columns[3:25])
+    df_nomic = df_nomic.drop(columns=df_nomic.columns[0:2])
+else:
+    df_nomic = None
+
+
+# In[7]:
+
+
+# subset each column that contains metadata
+metadata = data_df.filter(regex="Metadata")
+
+# get all columns that are not metadata except for metadata_Well
+data = data_df.drop(metadata.columns, axis=1)
+
+# get the metadata_Well column
+metadata_well = metadata["Metadata_Well"]
+
+data = pd.merge(data, metadata_well, left_index=True, right_index=True)
+
+
+# In[8]:
+
+
+if (aggregation == True) and (nomic == True):
+
+    # subset each column that contains metadata
+    metadata = data_df.filter(regex="Metadata")
+    data_df = data_df.drop(metadata.columns, axis=1)
+    data_df = pd.concat([data_df, metadata["Metadata_Well"]], axis=1)
+    # groupby well and take mean of each well
+    data_df = data_df.groupby("Metadata_Well").mean()
+    # drop duplicate rows in the metadata_well column
+    metadata = metadata.drop_duplicates(subset=["Metadata_Well"])
+    # get the metadata for each well
+    data_df = pd.merge(
+        data_df, metadata, left_on="Metadata_Well", right_on="Metadata_Well"
+    )
+    data_df = pd.merge(
+        data_df, df_nomic, left_on="Metadata_Well", right_on="position_x"
+    )
+    data_df = data_df.drop(columns=["position_x"])
+
+elif (aggregation == True) and (nomic == False):
+    # subset each column that contains metadata
+    metadata = data_df.filter(regex="Metadata")
+    data_df = data_df.drop(metadata.columns, axis=1)
+    data_df = pd.concat([data_df, metadata["Metadata_Well"]], axis=1)
+    # groupby well and take mean of each well
+    data_df = data_df.groupby("Metadata_Well").mean()
+    # drop duplicate rows in the metadata_well column
+    metadata = metadata.drop_duplicates(subset=["Metadata_Well"])
+    # get the metadata for each well
+    data_df = pd.merge(
+        data_df, metadata, left_on="Metadata_Well", right_on="Metadata_Well"
+    )
+elif (aggregation == False) and (nomic == True):
+    data_df = pd.merge(
+        data_df, df_nomic, left_on="Metadata_Well", right_on="position_x"
+    )
+    data_df = data_df.drop(columns=["position_x"])
+elif aggregation == False and nomic == False:
+    pass
+else:
+    print("Error")
```

Comment:
No need to change this code, but in the future it may be better to use `try, except` for code that may produce errors.